core/modules/widgets/navigator.js

```diff
@@ -561,6 +562,8 @@ NavigatorWidget.prototype.handlePerformImportEvent = function(event) {
 		importReport = [];
 	// Add the tiddlers to the store
 	importReport.push($tw.language.getString("Import/Imported/Hint") + "\n");
+	// Let the loggers do their work
+	importTiddler = $tw.hooks.invokeHook("th-before-importing",importTiddler);
```

Comment:
I recommend moving the variable assignment for importData from above to here.

`importData = this.wiki.getTiddlerDataCached(event.param,{tiddlers: {}})`

This would make the hook a lot more versatile as if needed, the raw JSON payload of the importTiddler could be manipulated.

Reply:
>This would make the hook a lot more versatile as if needed, the raw JSON payload of the importTiddler could be manipulated.

IMO at this point it's way too late to manipulate tiddlers. Import tiddlers can be manipulated using the "upgraders" mechanism. You can manipulate on a "per tiddler" basis. It's simple

Reply:
@pmario then your documentation and code are not coherent. If the code is to stay as is, your documentation update needs to reflect that the only the fields of the `importTiddler` other than the text field with the `JSON` data can be manipulated in the hook that you are adding. 

Furthermore, I don't really see a reason to artificially limit the utility of a hook in this manner. However, regardless of that please ensure that the docs update accurately reflects what the hook can and cannot be used for.

Reply:
done! See lines below